testing/test_functions.py

```diff
@@ -36,6 +31,8 @@ def test_next_tag(tag, expected):
 
 c = Configuration()
 
+# 2009-02-13T23:31:30+00:00
+os.environ["SOURCE_DATE_EPOCH"] = "1234567890"
```

Comment:
nice idea

please use `monkeypatch.setenv` inside of the test instead of doing a global patching there

else the behavior bleeds to other tests an

Reply:
yeah, I thought about that. Problem is that, AFAICT, the `meta()` objects in the `VERSIONS` array get created at global scope, so modifying the environment within the test case doesn't affect the `.time` attribute that's already set. That is, if I do
```
 # 2009-02-13T23:31:30+00:00
-os.environ["SOURCE_DATE_EPOCH"] = "1234567890"
+#os.environ["SOURCE_DATE_EPOCH"] = "1234567890"
 VERSIONS = {
     "exact": meta("1.1", distance=None, dirty=False, config=c),
     "zerodistance": meta("1.1", distance=0, dirty=False, config=c),
@@ -60,7 +60,8 @@ VERSIONS = {
         ("distancedirty", "post-release node-and-date", "1.1.post3+d20090213"),
     ],
 )
-def test_format_version(version, scheme, expected):
+def test_format_version(version, monkeypatch, scheme, expected):
+    monkeypatch.setenv("SOURCE_DATE_EPOCH", "1234567890")
     version = VERSIONS[version]
```
on top, the test fails (as I expected) with
```
E       AssertionError: assert '1.1.post3+d20201014' == '1.1.post3+d20090213'
E         - 1.1.post3+d20201014
E         ?               - ^^^
E         + 1.1.post3+d20090213
E         ?              +++  ^
```
I don't really know much about pytest (I can't even get the full test suite to run), so I'm at a loss of how to add a real test.

Reply:
hmm, conftest.py already sets one env variable globally (which then gets unset in a few tests). Perhaps SOURCE_DATE_EPOCH should be set there as well and used for all the date tests - after all, the `assert today.strftime(".d%Y%m%d") in wd.version in test_git.py has a microscopic chance of a false negative is the test is run very close to (UTC) midnight.

Reply:
Good idea,

For pedantic we ought to have a single test that checks the no env var case

Reply:
All good now?

Reply:
yup, nicely done, thanks